### PR TITLE
BLD: add travis, coveralls, and codecov.

### PR DESCRIPTION
**Pull request by CJ-Wright**
I am mostly opening this PR to try to trigger travis to make certain it is working.

### CODE REVIEW

**Comment by CJ-Wright** [comment]
We may need to add xpdAcq to Travis (it's free) if we want it to check the build upon a pull request.

**Comment by CJ-Wright** [comment]
This PR should be ready to go. One test is failing, see the [travis build](https://travis-ci.org/CJ-Wright/xpdAcq/jobs/158340291#L849), but I am not certain what the problem is exactly.

**Comment by sbillinge** [comment]
@CJ-Wright most of us have not worked extensively with travis.  I think that once it is set up, we do not interact any differently with GitHub, but it might be good if we add a few words to our git workflow docs (which I have no idea where they are, so we need to come up with a way for someone slow and forgetful like me to keep finding them) about it before we pull this in and close it.

In an vain effort to be able to find things related to group activities, I am starting to as people to links to things in the billinge-group google group, then I know that I can always go there to basically search for and find everything....

@chiahaoliu we don't need to pass that failing test to pull in the PR, but we should investigate why it is failing and fix it.  I think it is maybe best if you do that?

**Comment by CJ-Wright** [comment]
I will add some documentation about travis and coveralls (once I find the git workflow).
@chiahaoliu If it is easier I have also put the travis stuff into my copy of your refactoring branch, if it is easier I can PR into that so you have travis integration in your PR.

**Comment by sbillinge** [comment]
maybe this is a good time to revisit our git workflow!  

Rather than chatting about it in circles, why don't @CJ-Wright and @chiahaoliu rework the workflow docs, modifying anything that you think needs modifying, and we can build a discussion around that.

I gues the thing I would like to revisit a bit is how we handle branches.  Do we move to having a production branch and a development branch, or stay with the single master branch.  A quick review of how bluesky is done in detail would be helpful.

**Comment by tacaswell** [comment]
You should have a look at https://scikit-beam.github.io/scikit-beam/resource/dev_guide/index.html

**Comment by sbillinge** [comment]
I guess that issue can be closed by copy pasting that link into our google group so I can find it again in the future! ;)  Thanks @tacaswell 

btw, just to check....you guys are following those guidelines as written (more or less)?

**Comment by danielballan** [comment]
Yep. I just took a quick look at it again to be sure. It's also the way we are accustomed to operating with <name any scientific Python project with a medium-to-large community>. A more succinct and illustrated explanation of this workflow, called "git flow," is provided by github: https://guides.github.com/introduction/flow/

**Comment by chiahaoliu** [comment]
@sbillinge yes, as github work-flow, we are following the guidance listed in the link. So I guess we will still stay in the work-flow of having single master branch under upstream, issuing PR to that master branch under mainstream during development and tag master as a release once it is ready.

**Comment by danielballan** [comment]
I'll add that Travis doesn't change this workflow at all. It merely provides information: "If you merge this PR, the tests will (fail or pass)." One usually waits for the tests to pass to merge, but you always have the option to ignore Travis.

**Comment by tacaswell** [comment]
If you have a need for a bug fix branch, the pattern I suggest following is when you tag create a tag `vN.M.0` create a  `vN.M.x` branch to be keep as a bug fix branch.

**Comment by sbillinge** [comment]
Thanks @tacaswell.  @chiahaoliu , @CJ-Wright , please could we add this to our git flow.

@tacaswell do you branch the bugfix branch off the release branch?

**Comment by danielballan** [comment]
@sbillinge Since I saw this I'll answer for @tacaswell. You base it off of the master branch, starting from the tag. Like this:

```
# Tag a release.
git checkout master
git commit --allow-empty -m "REL: v0.4.0"  # an empty commit marking the release (optional)
git tag -a v0.4.0
git push upstream master
git push upstream --tags

# Start the bugfix branch for that version.
git checkout -b v0.4.x
git push upstream v0.4.x
```

**Comment by sbillinge** [comment]
I see.  It works best if you create it at the  same time you do the tag
even though you don't need it yet.....

**Comment by danielballan** [comment]
Right. Because it does not have any new commits (commits that will belong
to the next release) until you specifically "back port" some.

**Comment by chiahaoliu** [comment]
@sbillinge @CJ-Wright 
Just a quick update. The known fail from ``test_prun_with_bluesky`` actually comes from a historical issue. In this test, I used a detector object``det`` from ``bluesky.example`` to represent ``pe1`` when I wrote the test. At the point of time, ``det`` was still serializable, however for real ``pe1`` object at XPD is never serializable (yet) and in current version of ``bluesky``(the version Travis was testing with) ``det`` is not serializable either, which makes total sense.

In short, I think this PR is ready to go and we know the origin of that known fail
